### PR TITLE
Improvements

### PR DESCRIPTION
**Pull request by gbougeard**

### CODE REVIEW

File: src/main/scala/tv/teads/github/api/services/GithubService.scala [hunk 1]
```diff
@@ -61,6 +61,9 @@ abstract class GithubService(config: GithubApiClientConfig) extends LazyLogging
     customToken:  Option[String]      = None
   )(implicit ec: ExecutionContext, ev: Decoder[List[T]]): Future[List[T]] = {
     val httpUrl = HttpUrl.parse(s"${config.apiUrl}/$route").newBuilder().addAllParams(params).build()
+
+    logger.debug(s"fetchMultiple $httpUrl")
```
Comment (pdalpra) [inline src/main/scala/tv/teads/github/api/services/GithubService.scala]: doesn't help much :/

File: src/test/scala/tv/teads/github/api/services/IssueServiceSpec.scala [hunk 2]
```diff
@@ -38,4 +38,35 @@ class IssueServiceSpec extends BaseSpec {
...
+    whenReady(teadsClient.issues.fetchComments("github-api-client")) { list ⇒
+      list should not be empty
+    }
+  }
+
+  it should "be able to fetch a specific github-api-client issue comment" in {
+
+    val commentId = 150513228
+
+    whenReady(teadsClient.issues.fetchComment("github-api-client", commentId)) { list ⇒
```
Comment (pdalpra) [inline src/test/scala/tv/teads/github/api/services/IssueServiceSpec.scala]: not a list

File: src/main/scala/tv/teads/github/api/services/GithubService.scala (hunk 1 above)
Comment (gbougeard) [inline src/main/scala/tv/teads/github/api/services/GithubService.scala]: for debugging it does :D

File: src/main/scala/tv/teads/github/api/http/OkHttpClientWrapper.scala [hunk 3]
```diff
@@ -19,6 +20,7 @@ private[api] class OkHttpClientWrapper(authenticator: Option[Authenticator], max
   def executeAsync(requestBuilder: Request.Builder, customAuthenticator: Option[Authenticator]): Future[Response] = {
     val auth = customAuthenticator orElse authenticator
     val authenticatedRequest = auth.map(_(requestBuilder)).getOrElse(requestBuilder).build()
+    logger.debug(s"Calling Github : url ${authenticatedRequest.url()}")
```
Comment (pdalpra) [inline src/main/scala/tv/teads/github/api/http/OkHttpClientWrapper.scala]: add the method too, only the URL is not sufficient.
Adding the body could be useful too

File: src/test/scala/tv/teads/github/api/services/DeploymentServiceSpec.scala [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+  it should "be able to fetch github-api-client deployments" in {
+
+    whenReady(teadsClient.deployments.list("github-api-client", DeploymentFilter())) { list ⇒
+      list.foreach(d ⇒ println(s"${d.id} ${d.description}"))
+      list should not be empty
+    }
+  }
+
+  it should "create a deployment statuss" in {
```
Comment (pdalpra) [inline src/test/scala/tv/teads/github/api/services/DeploymentServiceSpec.scala]: typo

File: src/main/scala/tv/teads/github/api/services/DeploymentService.scala [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    task:        Option[String] = None,
+    environment: Option[String] = None
+  )
+
+  case class DeploymentParam(
+    ref:               String,
+    task:              Option[String]       = Some("deploy"),
+    auto_merge:        Option[Boolean]      = Some(true),
+    required_contexts: Option[List[String]] = None,
+    payload:           Option[Json]         = None,
```
Comment (pdalpra) [inline src/main/scala/tv/teads/github/api/services/DeploymentService.scala]: I'd prefer a string, to not force circe to github-api-client users

File: src/main/scala/tv/teads/github/api/model/Deployment.scala [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+  self: UserCodec with ZonedDateTimeCodec ⇒
+
+  implicit lazy val deploymentDecoder = Decoder.instance { cursor ⇒
+    for {
+      url ← cursor.downField("url").as[String]
+      id ← cursor.downField("id").as[Long]
+      sha ← cursor.downField("sha").as[String]
+      ref ← cursor.downField("ref").as[String]
+      task ← cursor.downField("task").as[String]
+      payload ← cursor.downField("payload").as[Option[Json]]
```
Comment (pdalpra) [inline src/main/scala/tv/teads/github/api/model/Deployment.scala]: I'd prefer a string, to not force circe to github-api-client users

**Comment by gbougeard** [comment]
ping

File: src/main/scala/tv/teads/github/api/http/OkHttpClientWrapper.scala [hunk 7]
```diff
@@ -2,13 +2,14 @@ package tv.teads.github.api.http
...
 
+import com.typesafe.scalalogging.LazyLogging
 import okhttp3.OkHttpClient.Builder
 
 import scala.concurrent.{Promise, Future}
 
 import okhttp3._
 
-private[api] class OkHttpClientWrapper(authenticator: Option[Authenticator], maxCacheSize: Long, cacheRoot: File) {
+private[api] class OkHttpClientWrapper(authenticator: Option[Authenticator], maxCacheSize: Long, cacheRoot: File) extends LazyLogging {
```
Comment (pdalpra) [inline src/main/scala/tv/teads/github/api/http/OkHttpClientWrapper.scala]: StrictLogging

**Comment by pdalpra** [comment]
lgtm

**Pull request merged**